### PR TITLE
Update branching doc

### PR DESCRIPTION
**Pull request by amyblais**

### CODE REVIEW

File: site/content/contribute/getting-started/branching.md [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
-description: "Mattermost recently moved away from a tick-tock release strategy with alternating feature and quality releases."
+title: "Mattermost Cherry-Pick process"
+heading: "Mattermost Cherry-Pick Process"
+description: "What the branching strategy and cherry-pick process looks like."
 date: 2019-06-18T00:00:00-04:00
 weight: 20
 ---
 
-Mattermost previously adopted a [tick-tock release strategy](https://docs.mattermost.com/process/release-faq.html#release-overview) where every other release was a "quality release" that only had bug fixes and no new features.
+The self-managed releases are now cut based off of the Mattermost Cloud release tags (e.g Mattermost Server v6.3 release was based off of ``cloud-2021-12-08-1`` Cloud release tag). See [the Hanbook release process](https://handbook.mattermost.com/operations/research-and-development/product/release-process/release-overview#cloud-release-branch-processes) for more details.
```
Comment (enahum) [inline site/content/contribute/getting-started/branching.md:9]: I believe this is not true for all repos, right ?

**Review by justinegeffen — APPROVED** [review]

File: site/content/contribute/getting-started/branching.md (hunk 1 above)
Comment (amyblais) [reply site/content/contribute/getting-started/branching.md]: I've clarified the repos.

**Comment by enahum** [comment]
is this PR missing the branching strategy for mobile?

**Comment by amyblais** [comment]
I've added a line for Mobile and Desktop.

**Review by cwarnermm — APPROVED** [review]

**Pull request merged**